walkbox.go

```diff
@@ -0,0 +1,105 @@
+package pctk
+
+import "log"
+
+const (
+	// MinPolygonVertices defines the minimum number of vertices to form a polygon.
+	MinPolygonVertices = 3
+)
+
+// Walkbox refers to a convex polygonal area that defines the walkable space for actors.
+type WalkBox struct {
+	WalkBoxID string
+	Enabled   bool
+	Vertices  []*Position
+	// TODO: Scale float32
+}
+
+// NewWalkBox creates a new WalkBox with the given ID and vertices.
+// It ensures the polygon formed by the vertices is convex. If not, it will cause a panic.
+// Why convex? Because you can draw a straight line/path between any two vertices inside the polygon
+// without needing to implement complex pathfinding algorithms.
+func NewWalkBox(id string, vertices []*Position) *WalkBox {
```

Comment:
I think it would be wise to limit the number of vertices to 4. Otherwise, the computation will be really hard. 

This can be done either declaring the input type as `[4]*Position`, or checking the length of the slice is 4 and failing otherwise as we do for non convex forms.